web/app/Services/Wikitext/HtmlOutput.php

```diff
@@ -3,27 +3,45 @@
 
 namespace Wikijump\Services\Wikitext;
 
-use \FFI;
-
-/**
- * Class HtmlOutput, representing a returned 'struct ftml_html_output' object.
- * @package Wikijump\Services\Wikitext
- */
 class HtmlOutput
 {
+    /**
+     * @var string The HTML output of the rendering process.
+     *
+     * This is not a complete HTML document, but only the portion rendered from the syntax tree.
+     * The recipient is responsible for forming this into a full HTML document.
+     */
     public string $html;
+
+    /**
+     * @var string The CSS output of the rendering process.
+     *
+     * This concatenates separate CSS styles specified in the tree in one string.
+     * The styles are not necessarily valid or safe, as they come from whatever the user inputted.
+     */
     public string $style;
```

Comment:
Out of scope for this PR, but noting that I think it'd be helpful to retain Wikidot's behaviour of putting each CSS snippet into separate style elements - I've found that it's helpful for introspection, and it also means that the end of a CSS file being badly-formatted (e.g. missing end brace) won't break all subsequent CSS, and in the future it might make it easier to reduce CSS duplication from `component:`s by checking for style elements with identical contents.

Actually thinking about it, I think splitting the CSS is outright necessary unless we want to add some magic to move all `@import` statements to the top of the concatenated file

Reply:
I guess that's fair. I originally combined them to avoid extra string buffer allocations in Rust, but that's kind of moot given all the FFI and such happening.